Zend/zend_language_scanner.l

```diff
@@ -1647,8 +1647,14 @@ NEWLINE ("\r"|"\n"|"\r\n")
 <ST_IN_SCRIPTING>{LNUM} {
 	char *end;
 	if (yyleng < MAX_LENGTH_OF_LONG - 1) { /* Won't overflow */
+		/* Zero-terminate the recognized LNUM before parsing
+		 * so that Windows will produce the correct parse error.
+		 */
+		char save = yytext[yyleng];
+		yytext[yyleng] = '\0';
```

Comment:
I don't think that modifying yytext is legal.

Reply:
Is there a better approach? @cmb69 suggested this patch, and it seems to work.

Reply:
Passing 10 instead of 0 as base will probably do it.

Reply:
...with some more explicit treatment for octal.

Reply:
What would the more explicit octal treatment look like?

Reply:
Using `ZEND_STRTOL(yytext, &end, yytext[0] == '0' ? 8 : 10)` would probably be enough.

Reply:
I updated the patch and verified that it still works.